dispatcher/DAQController.py

```diff
@@ -89,160 +87,92 @@ def SolveProblem(self, latest_status, goal_state):
                 self.one_detector_arming = True
 
         '''
-        CASE 1: DETECTORS ARE INACTIVE
-
-        In our case 'inactive' means 'stopped'. An inactive detector is in its goal state as 
-        long as it isn't doing anything, i.e. not ARMING, ARMED, or RUNNING. We don't care if 
-        it's idle, or in error, or if there's no status at all. We will care about that later
-        if we try to activate it.
+        CASE 1: DETECTORS ARE INACTIVE (IDLE)
+        'Inactive' means 'stopped'. An inactive detector is in its goal state as long as it isn't doing anything, i.e. it is in neither of the active_states.
         '''
-        # 1a - deal with TPC and also with MV and NV, but only if they're linked
-        active_states = [STATUS.ARMING, STATUS.ARMED, STATUS.RUNNING,
-                         STATUS.ERROR, STATUS.UNKNOWN]
-        if goal_state['tpc']['active'] == 'false':
-
-            # Send stop command if we have to
-            if (
-                    # TPC not in Idle, error, timeout
-                    (latest_status['tpc']['status'] in active_states) or
-                    # MV linked and not in Idle, error, timeout
-                    (latest_status['muon_veto']['status']  in active_states and
-                     goal_state['tpc']['link_mv'] == 'true') or
-                    # NV linked and not in Idle, error, timeout
-                    (latest_status['neutron_veto']['status']  in active_states and
-                     goal_state['tpc']['link_nv'] == 'true')
-            ):
-                self.StopDetectorGently(detector='tpc')
-            elif latest_status['tpc']['status'] == STATUS.TIMEOUT:
-                self.CheckTimeouts('tpc')
-
-        # 1b - deal with MV but only if MV not linked to TPC
-        if goal_state['tpc']['link_mv'] == 'false' and goal_state['muon_veto']['active'] == 'false':
-            if latest_status['muon_veto']['status'] in active_states:
-                self.StopDetectorGently(detector='muon_veto')
-            elif latest_status['muon_veto']['status'] == STATUS.TIMEOUT:
-                self.CheckTimeouts('muon_veto')
-        # 1c - deal with NV but only if NV not linked to TPC
-        if goal_state['tpc']['link_nv'] == 'false' and goal_state['neutron_veto']['active'] == 'false':
-            if latest_status['neutron_veto']['status'] in active_states:
-                self.StopDetectorGently(detector='neutron_veto')
-            elif latest_status['neutron_veto']['status'] == STATUS.TIMEOUT:
-                self.CheckTimeouts('neutron_veto')
+        active_states = [STATUS.RUNNING, STATUS.ARMED, STATUS.ARMING, STATUS.UNKNOWN]
 
+        for det in latest_status.keys():
+            # The detector should be INACTIVE
+            if goal_state['tpc']['active'] == 'false':
```

Comment:
I think we did see this during the test, should it say 'tpc' or detector as the first key?

Reply:
Right, now is changed